WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -92,11 +99,47 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
     @Published private(set) var targetLanguageText: String = ""
     @Published private(set) var targetDeviceText: String = ""
 
+    // AI Suggestions
+    @Published private(set) var isLoadingAISuggestions: Bool = true
+    private let storage: StorageManagerType
+    private var product: Product? {
+        guard let product = productsResultsController.fetchedObjects.first else {
+            assertionFailure("Unable to fetch product with ID: \(productID)")
+            return nil
+        }
+        return product
+    }
+
+    @Published private(set) var errorState: ErrorState = .none
+    private var suggestions: [BlazeAISuggestion] = []
+
+    var canConfirmDetails: Bool {
+        guard !isDownloadingImage else {
+            return false
+        }
+        return tagline.isNotEmpty && description.isNotEmpty
+    }
+
+    /// ResultController to to track the current product count.
+    ///
+    private lazy var productsResultsController: ResultsController<StorageProduct> = {
+        let predicate = \StorageProduct.siteID == siteID && \StorageProduct.productID == productID
+        let controller = ResultsController<StorageProduct>(storageManager: storage, matching: predicate, sortedBy: [])
+        try? controller.performFetch()
```

Comment:
Should we log the error when the fetch fails?

Reply:
Added in 59a5b1a9ff6ec3234a285c10f91e3f9aaf25578c